tests/lib/PerFile/PerFile.cpp

```diff
@@ -84,64 +88,83 @@ void* PerFile_openandmap(const umt_optstruct_t* testops, uint64_t numbytes)
   if ( testops->iodirect ) 
     open_options |= O_DIRECT;
 
-  if ( !testops->noinit ) {
+  if ( testops->initonly || ( !testops->noinit && !testops->noio) ) {
     open_options |= O_CREAT;
     unlink(filename.c_str());   // Remove the file if it exists
   }
 
   handle->range_size = numbytes;
   handle->filename = filename;
-  if ( ( handle->fd = open(filename.c_str(), open_options, S_IRUSR | S_IWUSR) ) == -1 ) {
-    string estr = "Failed to open/create " + handle->filename + ": ";
-    perror(estr.c_str());
-    return NULL;
-  }
+  if ( !testops->noio ) {
+    if ( ( handle->fd = open(filename.c_str(), open_options, S_IRUSR | S_IWUSR) ) == -1 ) {
+      string estr = "Failed to open/create " + handle->filename + ": ";
+      perror(estr.c_str());
+      return NULL;
+    }
 
-  if ( open_options & O_CREAT ) { // If we are initializing, attempt to pre-allocate disk space for the file.
-    try {
-      int x;
-      if ( ( x = posix_fallocate(handle->fd, 0, handle->range_size) != 0 ) ) {
-        ostringstream ss;
-        ss << "Failed to pre-allocate " << handle->range_size << " bytes in " << handle->filename << ": ";
-        perror(ss.str().c_str());
+    if ( open_options & O_CREAT ) { // If we are initializing, attempt to pre-allocate disk space for the file.
+      try {
+        int x;
+        if ( ( x = posix_fallocate(handle->fd, 0, handle->range_size) != 0 ) ) {
+          ostringstream ss;
+          ss << "Failed to pre-allocate " << handle->range_size << " bytes in " << handle->filename << ": ";
+          perror(ss.str().c_str());
+          return NULL;
+        }
+      } catch(const std::exception& e) {
+        cerr << "posix_fallocate: " << e.what() << endl;
+        return NULL;
+      } catch(...) {
+        cerr << "posix_fallocate failed to allocate backing store\n";
         return NULL;
       }
-    } catch(const std::exception& e) {
-      cerr << "posix_fallocate: " << e.what() << endl;
-      return NULL;
-    } catch(...) {
-      cerr << "posix_fallocate failed to allocate backing store\n";
-      return NULL;
     }
-  }
 
-  struct stat sbuf;
-  if (fstat(handle->fd, &sbuf) == -1) {
-    string estr = "Failed to get status (fstat) for " + filename + ": ";
-    perror(estr.c_str());
-    return NULL;
-  }
+    struct stat sbuf;
```

Comment:
These lines appear to have extra white space in them.  Will investigate/fix.

Reply:
Also, the umapsort regression test is indeed passing.  However, I also just tried churn and it is not happy.  Will investigate as well.

Reply:
Actually, this part of the diff just shows that these lines had an extra indentation added to them which was intentional because they were moved within a new if block.